src/anonymous-credentials/presentation-builder.ts

```diff
@@ -764,4 +780,89 @@ export class PresentationBuilder extends Versioned {
       );
     }
   }
-}
+
+  private buildSignatureParams(
+    signature: Signature,
+    messageCount: number,
+    label: Uint8Array
+  ): SignatureParams {
+    if (signature instanceof BBSSignature) {
+      return BBSSignatureParams.generate(messageCount, label);
+    } else if (signature instanceof BBSPlusSignatureG1) {
+      return BBSPlusSignatureParamsG1.generate(messageCount, label);
+    } else if (signature instanceof PSSignature) {
+      return PSSignatureParams.generate(messageCount, label);
+    } else {
+      throw new Error(`Signature is invalid ${signature}`);
+    }
+  }
+
+  private buildSignatureSetupParam(
+    params: SignatureParams,
+    messageCount: number
+  ): SetupParam {
+    if (params instanceof BBSSignatureParams) {
+      return SetupParam.bbsSignatureParams(params.adapt(messageCount));
+    } else if (params instanceof BBSPlusSignatureParamsG1) {
+      return SetupParam.bbsPlusSignatureParamsG1(params.adapt(messageCount));
+    } else if (params instanceof PSSignatureParams) {
+      return SetupParam.psSignatureParams(params.adapt(messageCount));
+    } else {
+      throw new Error(`Signature params are invalid ${params}`);
+    }
+  }
+
+  private buildPublicKeySetupParam(
+    pk: BBSPublicKey | BBSPlusPublicKeyG2 | PSPublicKey,
+    messageCount: number
+  ): SetupParam {
+    if (pk instanceof BBSPublicKey) {
+      return SetupParam.bbsPlusSignaturePublicKeyG2(pk);
+    } else if (pk instanceof BBSPlusPublicKeyG2) {
+      return SetupParam.bbsPlusSignaturePublicKeyG2(pk);
+    } else if (pk instanceof PSPublicKey) {
+      const supported = pk.supportedMessageCount();
+      if (messageCount < pk.supportedMessageCount()) {
+        pk = pk.adaptForLess(messageCount)!;
+      } else {
+        throw new Error(`Unsupported message count: supported = ${supported}, received = ${messageCount}`);
+      }
+
+      return SetupParam.psSignaturePublicKey(pk as PSPublicKey);
+    } else {
+      throw new Error(`Public key is invalid ${pk}`);
+    }
+  }
+
+  private buildSignatureStatementFromParamsRef(
```

Comment:
This can be removed in favor of the method of the same name in `Presentation` if you pass public key instead of `signature`